scripts/run_mypy_checks.py

```diff
@@ -0,0 +1,793 @@
+    'scripts/linters/pylint_extensions_test.py',
+    'scripts/linters/python_linter.py',
+    'scripts/linters/python_linter_test.py',
+    'scripts/linters/warranted_angular_security_bypasses.py',
+    'scripts/pre_commit_hook.py',
+    'scripts/pre_commit_hook_test.py',
+    'scripts/pre_push_hook.py',
+    'scripts/pre_push_hook_test.py',
+    'scripts/regenerate_requirements.py',
+    'scripts/regenerate_requirements_test.py',
+    'scripts/release_scripts/cut_release_or_hotfix_branch.py',
+    'scripts/release_scripts/cut_release_or_hotfix_branch_test.py',
+    'scripts/release_scripts/repo_specific_changes_fetcher.py',
+    'scripts/release_scripts/repo_specific_changes_fetcher_test.py',
+    'scripts/release_scripts/update_changelog_and_credits.py',
+    'scripts/release_scripts/update_changelog_and_credits_test.py',
+    'scripts/release_scripts/update_configs.py',
+    'scripts/release_scripts/update_configs_test.py',
+    'scripts/run_backend_tests.py',
+    'scripts/run_custom_eslint_tests.py',
+    'scripts/run_e2e_tests.py',
+    'scripts/run_e2e_tests_test.py',
+    'scripts/run_frontend_tests.py',
+    'scripts/run_lighthouse_tests.py',
+    'scripts/run_mypy_checks.py',
+    'scripts/run_mypy_checks_test.py',
+    'scripts/run_portserver.py',
+    'scripts/run_presubmit_checks.py',
+    'scripts/run_tests.py',
+    'scripts/script_import_test.py',
+    'scripts/scripts_test_utils.py',
+    'scripts/scripts_test_utils_test.py',
+    'scripts/servers.py',
+    'scripts/servers_test.py',
+    'scripts/setup.py',
+    'scripts/setup_gae.py',
+    'scripts/setup_gae_test.py',
+    'scripts/setup_test.py',
+    'scripts/start.py',
+    'scripts/third_party_size_check.py',
+    'scripts/typescript_checks.py',
+    'scripts/typescript_checks_test.py',
+]
+
+
+CONFIG_FILE_PATH = os.path.join('.', 'mypy.ini')
+# TODO(#13113): Change mypy command to mypy path after Python3 migration.
+MYPY_CMD = 'mypy'
+MYPY_REQUIREMENTS_PATH = os.path.join('.', 'mypy_requirements.txt')
+PYTHON3_CMD = 'python3'
+
+
+_PARSER = argparse.ArgumentParser(
+    description='Type checking script for Oppia codebase.'
+)
+
+_PARSER.add_argument(
+    '--files',
+    help='Files to type-check',
+    action='append',
+    nargs='+'
+)
+
+
+def get_mypy_cmd(files):
+    """Return the appropriate command to be run."""
+    if files:
+        cmd = [MYPY_CMD, '--config-file', CONFIG_FILE_PATH] + files[0]
+    else:
+        excluded_files_regex = (
+            '|'.join(NOT_FULLY_COVERED_FILES + EXCLUDED_DIRECTORIES))
+        cmd = [
+            MYPY_CMD, '--exclude', excluded_files_regex,
+            '--config-file', CONFIG_FILE_PATH, '.'
+        ]
+    return cmd
+
+
+def install_mypy_prerequisites():
+    """Install mypy and type stubs from mypy_requirements.txt."""
+    cmd = [PYTHON3_CMD, '-m', 'pip', 'install', '-r', MYPY_REQUIREMENTS_PATH]
+    process = subprocess.call(
+        cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+    return process
```

Comment:
installing typing module is required. We are already using typing module. But now, I have added it to the PREREQUISITES list which has packages which need to be downloaded before installing other backend packages and importing modules. See here: https://github.com/oppia/oppia/blob/173ca903584c84f6e70abe46144d6e66d72b092e/scripts/install_third_party_libs.py#L37

Reply:
Sorry, I didn't get it! You mean we only need mypy + typings to be installed to run this script? In that case, do we need to run intall_third_party_libs in the `.github/workflows/backend_type_checks.yml `?

Reply:
Typing will be installed from `install_third_party_libs.py` and mypy will be installed from run_mypy_checks. So yes, `install_third_party_libs.py` is needed.

Reply:
So I would suggest installing everything through `install_third_party_libs` and running `install_third_party_libs` from `run_mypy_checks`. [Considering this check will raise error without initially running the `install_third_party_libs` script, right?]

Reply:
We tried this. But weren't able to do so because of py2 and py3 mismatch. So this was our solution till the python3 migration. A lot of problems are arising here due to current working in py2 version.  We cannot install everything from install_third_party_libraries right now. Thanks a lot!

Reply:
@hardikkat24 I think you didn't get what @DubeySandeep is trying to say here.

Currently all scripts can work based on the dependencies that were installed using `install_third_party_libs`. Take `run_backend_tests` for example -  it simply [calls third party libs script](https://github.com/oppia/oppia/blob/976a2146c250024a74359048686325d318b8a23f/scripts/run_backend_tests.py#L69) to ensure everything is setup before you can run it. 

Can't we move `install_mypy_prerequisites` function also to the `install_third_party_libs` file? You can write line 759 there too, right? This is a good structure we try to follow in the codebase and makes sure everything is setup before checks are run.

Currently if someone has a missing dependency and they try to run this check, they will get missing libraries errors instead of them getting automatically installed. Do you get what I am saying?

Reply:
Installing mypy prerequisites cannot be added to install_third_party_libs because mypy prerequisites are python3 modules. 
@sajalasati As per the discussion, I have added install_third_party_libs in the mypy script itself. Thanks!

Reply:
@DubeySandeep PTAL. I have added `install_third_party_libs` inside mypy script itself. PTAL.